### PR TITLE
feat(fastrp_embedding): add fast rp embeddings to xgboost

### PR DESCRIPTION
**Pull request by parkererickson-tg**

### CODE REVIEW

**Review by billshitg — APPROVED** [review]
The result seems too good to be true lol. But numbers don't lie. Only question is: do we want to keep the first cell that reinstalls pyTG?

**Comment by parkererickson-tg** [comment]
Oh, it is because I am using the callback in the dataloader. The result was very handy (I did have to tune the GNN a tiny bit). Will not merge until new version of pyTG is released.

**Comment by zhixiany** [comment]
When can we merge this one? @parkererickson-tg 

This is great to show the value of embedding for improving model accuracy. one minor comment: in addition to show sum of all embeddings in terms of importances, also good to show top X (say X=5) embedding's value as well.

**Comment by parkererickson-tg** [comment]
I am waiting on the e2e testing pipeline to work in order to create test cases and merge this. By top 5, do you mean by importance? There isn't that much of an interpretation to the embedding compared to something produced by PCA, for example.

**Review by qe-tigergraph — COMMENTED** [review]
Unit Test: SKIPPED, e2e Test: SUCCESS, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/612/

**Review by qe-tigergraph — APPROVED** [review]
QE Approved
